tools/slang-test/render-api-util.h

```diff
@@ -0,0 +1,61 @@
+#ifndef SLANG_RENDER_API_UTIL_H
+#define SLANG_RENDER_API_UTIL_H
+
+#include "../../source/core/slang-string.h"
+#include "../../source/core/slang-result.h"
+
+
+enum class RenderApiType
+{
+    kUnknown = -1,
+    kOpenGl = 0,
+    kVulkan,
+    kD3D12,
+    kD3D11,
+    kCountOf,
+};
+
+// Use a struct wrapped Enum instead of enum class, cos we want to be able to manipulate as integrals
+struct RenderApiFlag
+{
+    enum Enum
+    {
+        kOpenGl = 1 << int(RenderApiType::kOpenGl),
+        kVulkan = 1 << int(RenderApiType::kVulkan),
+        kD3D12 = 1 << int(RenderApiType::kD3D12),
+        kD3D11 = 1 << int(RenderApiType::kD3D11),
+        kAllOf = (1 << int(RenderApiType::kCountOf)) - 1                   ///< All bits set
+    };
+};
+
+struct RenderApiUtil
+{
+    struct Info
+    {
+        RenderApiType type;               ///< The type
+        const char* names;          ///< Comma separated list of names associated with the type
+    };
+
+        /// Returns true if the API is available. 
```

Comment:
This isn't a spaces/tabs thing, AFAICT, so I'm wondering why the documentation comments here are indented. Is that a convention of some kind?

Reply:
Oh you mean the comments above a function. The /// is for doxygen style comments, the indenting style is from a previous companys style. It's a small thing, but makes it a little easier to scan signatures from comment blocks.